pkg/volume/empty_dir/quota.go

```diff
@@ -0,0 +1,128 @@
+package empty_dir
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// An object for applying quota to an empty dir volume.
+type quotaApplicator interface {
+	// Apply the quota to the given EmptyDir path:
+	Apply(dir string, medium api.StorageMedium, pod *api.Pod) error
+}
+
+type xfsQuotaApplicator struct {
+}
+
+func (xqa *xfsQuotaApplicator) Apply(dir string, medium api.StorageMedium, pod *api.Pod, fsGroup *int64, quota int) error {
+	if medium == api.StorageMediumMemory {
+		glog.V(3).Infof("skipping quota application due to memory storage medium.")
+		return nil
+	}
+	if !xqa.isXfs(dir) {
+		glog.V(3).Infof("%s is not on an XFS filesystem, skipping quota application.")
```

Comment:
Seems like maybe this needs to be smarter, otherwise emptyDir on a misconfigured node would allow all pods to use unrestricted storage. I would expect failure to apply quota to propagate back up as a failure to establish the volume for the pod (except if the pod is somehow special and allowed to bypass quota application).